src/node_lkl.cc

```diff
@@ -150,3 +150,38 @@ NAN_METHOD(syscall) {
 		}
 	}
 }
+
+NAN_METHOD(parseDirent64) {
+	// Args:
+	// * Buffer buffer
+	// * int nread
+	// * Array result
+	// Parses buffer (which should be a lkl_linux_dirent64*) and fills the
+	// result array with filenames it finds.
+	// Filenames are Buffers that need to be decoded by the caller.
+	char* buf = (char*) node::Buffer::Data(info[0]);
+	int nread = info[1]->IntegerValue();
+	v8::Local<v8::Array> result = info[2].As<v8::Array>();
+
+	int posInResult = result->Length();
+	int bpos;
+	lkl_linux_dirent64 *dir_entry;
+
+	for (bpos = 0; bpos < nread;) {
+		dir_entry = (lkl_linux_dirent64 *) (buf + bpos);
```

Comment:
nread is supposed to be smaller than the buffer length.
I added 
```c++
assert(nread < node::Buffer::Length(info[0]));
```
in the beginning.